src/utils.rs

```diff
@@ -167,3 +168,69 @@ impl<'a, T, const N: usize> IntoOption<&'a [T]> for &'a [T; N] {
         Some(self)
     }
 }
+
+pub trait ScalarOrArray<'a> {
+    type Array: AsRef<Array> + 'a;
+
+    fn into_owned_or_ref_array(self) -> Self::Array;
+}
+
+impl<'a> ScalarOrArray<'a> for Array {
+    type Array = Array;
+
+    fn into_owned_or_ref_array(self) -> Array {
+        self
+    }
+}
+
+impl<'a> ScalarOrArray<'a> for &'a Array {
+    type Array = &'a Array;
+
+    // TODO: clippy would complain about `as_array`. Is there a better name?
```

Comment:
`as_mlx_array`?

Reply:
clippy actually complains about the use of `as_` when the argument is `self` instead of `&self`